### PR TITLE
Attitude model

### LINKED ISSUE #193: Attitude model
# Feature
this issue cover point 2 and part of 3 in #190 

## Desired Behavior / Functionality
Adding a spacecraft attitude model specifying initial attitude conditions and attitude evolution through time (with external attitude disturbances). 

## What Needs to Be Done
Create an attitude model class with initial conditions. 
Specify body fixed pointing vector (used for plotting, testing and orientation).
Add "update attitude" function, updating spacecraft state (roll, pitch, yaw angles, pointing vector orientation, angular velocity and acceleration).

## How Can It Be Tested
Check pointing vector evolution with zero initial angular velocity, angular velocities around specific axes testing time between full rotations. Do the same with (constant) angular acceleration.

### PR DESCRIPTION
**Pull request by Mr-Medina**
# Description
An attitude model is added, simulating the evolution of a spacecraft actor attitude through time with external attitude disturbances. User can specify initial contitions: spacecraft attitude (being roll, pitch and yaw angles) and angular velocity. 
Disturbance torque calulations are not implemented yet but framework is present for later implementation.

Reference frame transfer file is also made in utils, able to switch between Earth-centered inertial, roll pitch yaw (based on nadir & along track directions) and body fixed frames. Also used for vector rotations in reference frames and extracting roll, pitch, and yaw angles between frames. 
Summary of changes

* attitude model added
* reference frame transformation functions added
* disturbance calculations framework added

## Resolved Issues

- [ ] fixes #193 

## How Has This Been Tested?
I made a file which plots the pointing vector in the spacecraft trajectory using matplotlib. I was not sure if I needed to add this in the PR. Please refer to my fork to file "test_attitude_plotting.py" for visual reference.
Further testing methods will be added after feedback is received.

### CODE REVIEW

**Comment by gomezzz** [comment]
@Mr-Medina Should we have a look and give you some feedback? :) 

You can request a review in the top right with

grafik

File: paseos/attitude/attitude_model.py [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        # principal axes:
+        x = body_to_rpy(np.array([1, 0, 0]), self._actor_attitude_in_rad)
+        y = body_to_rpy(np.array([0, 1, 0]), self._actor_attitude_in_rad)
+        z = body_to_rpy(np.array([0, 0, 1]), self._actor_attitude_in_rad)
+
+        # pointing vector:
+        p = body_to_rpy(self._actor_pointing_vector_body, self._actor_attitude_in_rad)
+        return x, y, z, p
+
+    def update_attitude(self, dt):
```
Comment (GabrieleMeoni) [inline paseos/attitude/attitude_model.py:177]: Make sure this function is always called inside PASEOS main loop to be sure the variables will be updated all at the same time: https://github.com/aidotse/PASEOS/blob/e5bb071b4ee2988c2765497aa8fe3d0f6eacb655/paseos/paseos.py#L123

**Review by GabrieleMeoni — CHANGES_REQUESTED** [review]
At a first glance, it looks good. We can have a dedicated discussion later on. I would wait for  the geometric model PR to be updated as requested. This code needs also to capitalize on that PR.

File: paseos/attitude/attitude_model.py [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        return -u / np.linalg.norm(u)
+
+    def calculate_disturbance_torque(self):
+        """Compute total torque due to user specified disturbances
+
+        Returns:
+            np.array([Tx, Ty, Tz]): total combined torques in Nm
+        """
+
+        T = np.array([0.0, 0.0, 0.0])
```
[the remark this answers is not in the document]
Comment (Mr-Medina) [reply paseos/attitude/attitude_model.py:86]: Yes, the T vector will be expressed in the body frame

File: paseos/attitude/attitude_model.py [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+        return T
+
+    def calculate_angular_acceleration(self):
+        """Calculate the spacecraft angular acceleration (external disturbance torques and gyroscopic accelerations)"""
+        # todo: implement geometric model
+        # I = self._actor_moment_of_inertia
+        I = np.array([[50, 0, 0], [0, 50, 0], [0, 0, 50]])  # test placeholder
+
+        # Euler's equation for rigid body rotation: a = I^(-1) (T - w x (Iw))
+        # a = angular acceleration, I = inertia matrix, T = torque vector, w = angular velocity
```
[the remark this answers is not in the document]
Comment (Mr-Medina) [reply paseos/attitude/attitude_model.py:103]: done!

**Review by Moanalengkeek — APPROVED** [review]

**Pull request merged**